### PR TITLE
Allow customizing through relations local keys

### PR DESCRIPTION
**Pull request by muhammedkamel**
There is no way to customize the local keys of the through relations.

Imagine the following scenario:

Model A
- HasMany B

Model C
- HasOne B

Model B
- BelongsTo C
- BelongsTo B

belongs-to-through

From Model `C` I need to access Model `A` using `belongsToThrough` relation using `B.c_id` and `B.a_id`

The new implementation should enable this using the following

```php
<?php

class C extends Model
{
    public function a(): BelongsToThrough
    {
        return $this->belongsToThrough(
            A::class,
            B::class,
            foreignKeyLookup: [B::class => 'id'],
            localKeyLookup: [B::class => 'c_id'],
        );
    }
}

```

and using this query
```sql
select "A".* from "A" inner join "B" on "B"."a_id" = "A"."id" where "B"."c_id" = 416 limit 1
```

### CODE REVIEW

**Comment by muhammedkamel** [comment]
@staudenmeir any update?!

File: src/Relations/BelongsToThrough.php [hunk 1]
```diff
@@ -121,7 +138,24 @@ public function getForeignKeyName(Model $model = null)
...
+     */
+    public function getLocalKeyName(Model $model = null): string
+    {
+        $table = explode(' as ', ($model ?? $this->parent)->getTable())[0];
+
+        if (array_key_exists($table, $this->localKeyLookup)) {
+            return $this->localKeyLookup[$table];
+        }
+
+        return 'id';
```
Comment (staudenmeir) [inline src/Relations/BelongsToThrough.php:158]: This is a breaking change: The fallback needs to be `getQualifiedKeyName()` as it was before (https://github.com/staudenmeir/belongs-to-through/blob/master/src/Relations/BelongsToThrough.php#L96)

Comment (muhammedkamel) [reply src/Relations/BelongsToThrough.php]: You are 100% correct, I have fixed it.

**Comment by staudenmeir** [comment]
Thanks! I released a new version.

**Pull request merged**